src/main/java/christmas/controller/EventPlannerController.java

```diff
@@ -0,0 +1,130 @@
+import christmas.utils.exception.ExceptionHandler;
+import christmas.utils.validators.DateValidator;
+import christmas.utils.validators.OrderItemValidator;
+import christmas.utils.validators.OrderValidator;
+import christmas.view.Input;
+import christmas.view.OutputView;
+import java.util.List;
+import java.util.Map;
+
+public class EventPlannerController {
+    private final Input inputView;
+    private final InputConvertor convertor;
+    private DiscountService discountService;
+    private BenefitService benefitService;
+    private Order order;
+    private List<OrderItem> orderItems;
+    private Date date;
+    private PayAmount payAmount;
+
+    private EventPlannerController(final Input inputView, final InputConvertor convertor) {
+        this.inputView = inputView;
+        this.convertor = convertor;
+    }
+
+    public static EventPlannerController of(final Input input, final InputConvertor convertor) {
+        return new EventPlannerController(input, convertor);
+    }
+
+    public void run() {
+        OutputView.printHello();
+        guideDate();
+        guideOrder();
+        OutputView.printExpectEvent();
+        int totalOrderAmount = orderMenu();
+        payAmount = new PayAmount(totalOrderAmount);
+        requestTotalOrder();
+        discountService = DiscountService.from(payAmount, date);
+        benefitService = BenefitService.from(payAmount);
+        discount();
+        int totalDiscountAmount = sale();
+        total();
+        badgy(totalDiscountAmount);
+    }
+
+    private void guideDate() {
+        String day = inputView.readDate();
+        date = ExceptionHandler.convert(convertor::convertDate, day, new DateValidator());
+        if (date == null) {
+            guideDate();
+        }
+    }
+
+    private void guideOrder() {
+        guideOrderItems();
+        order = ExceptionHandler.convert(convertor::convertOrder, orderItems, new OrderValidator());
+        if (order == null) {
+            guideOrder();
+        }
+    }
+
+    private void guideOrderItems() {
+        String orders = inputView.readOrder();
+        orderItems = ExceptionHandler.convert(convertor::convertOrderItems, orders,
+                new OrderItemValidator());
+        if (orderItems == null) {
+            guideOrderItems();
+        }
+    }
+
+    private int orderMenu() {
+        OrderOuputDto orderOuputDto = OrderOuputDto.fromOrder(order);
+        OutputView.printOrderMenu(orderOuputDto);
+        return order.requestTotalPrice();
+    }
+
+    private void requestTotalOrder() {
+        PayAmountDto payAmonutDto = PayAmountDto.fromPayAmonut(payAmount);
+        OutputView.printTotalPayAmount(payAmonutDto);
+    }
+
+    private void discount() {
+        int mainCount = order.requestMainCount();
+        int dessertCount = order.requestDessertCount();
+        Benefit benefit = benefitService.getBenefit();
+        discountService.discount(mainCount, dessertCount, benefit.isGift());
```

Comment:
```suggestion
        Benefit benefits = benefitService.getBenefits();
        discountService.discount(mainCount, dessertCount, benefits);
```
요렇게하고 증정품 유무를 서비스 안에서 판단하는게 좋지 않을까요? 그리고 복수형인게 확장성이 더 좋아보이네요